src/maingoblint.ml

```diff
@@ -145,6 +145,8 @@ let check_arguments () =
   if get_bool "ana.autotune.enabled" && get_bool "incremental.load" then (set_bool "ana.autotune.enabled" false; warn "ana.autotune.enabled implicitly disabled by incremental.load");
   if get_bool "exp.basic-blocks" && not (get_bool "justcil") && List.mem "assert" @@ get_string_list "trans.activated" then (set_bool "exp.basic-blocks" false; warn "The option exp.basic-blocks implicitely disabled by activating the \"assert\" tranformation.");
   if (not @@ get_bool "witness.invariant.all-locals") && (not @@ get_bool "cil.addNestedScopeAttr") then (set_bool "cil.addNestedScopeAttr" true; warn "Disabling witness.invariant.all-locals implicitly enables cil.addNestedScopeAttr.");
+  if (get_int "exp.unrolling-factor" > 0 || AutoTune0.isActivated "loopUnrollHeuristic") && (get_bool "witness.yaml.enabled" || get_string "witness.yaml.validate" <> "" || get_string "witness.yaml.unassume" <> "") then
+    fail "YAML witnesses are incompatible with syntactic loop unrolling (https://github.com/goblint/analyzer/pull/1370).";
```

Comment:
I think this is not a smart choice to make for now. At the very least we should extensively benchmark that impacts this has. I fear disabling loop unrolling will massively harm us in SV-COMP. This should be extensively benchmarked before changing it for our SV-COMP runs. (Ideally in a separate PR where this can be discussed extensively).

Reply:
We already discussed this at GobCon and decided to go this route to be able to make any progress with Freiburg. SV-COMP 2025 is further away, so #1370 is less urgent at this point.

It is impossible to separate these location fixes from this incompatibility: the fixed loop identification is based on `Loop` statements, which the syntactic unrolling doesn't have for the unrolled iterations, otherwise our witnesses are simply unsound (#1225).

For our own sv-benchmarks runs with only verification, no validation, we could flip the choice, but that is inherently unrepresentative of actual SV-COMP with validation. If we output a trivially falsifiable loop invariant `i == 0 && i == 1 && ...` for a loop that doesn't even need unrolling (for which `0 <= i` would be valid), then we've just shot ourselves in the foot.

There's no way around #1370. The question is simply whether we block this PR and the Freiburg collaboration behind #1370 or not (and do it later by the time for SV-COMP 2025).

Reply:
This is now reverted after #1403. That seems to work for now.